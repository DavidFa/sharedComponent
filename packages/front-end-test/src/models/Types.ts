export interface PostType {
    id: number;
    title: string;
    body: string
}

export enum PostStatus {
    list,
    edit,
    add
}