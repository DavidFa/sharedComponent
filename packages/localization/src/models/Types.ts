export type InputFormModel = {
    name: InputFormProps;
    description: InputFormProps;
}


export type InputFormProps = {
    label: string;
    placeholder: string;
    length: number;
}

export type InputFormWithLanguageProps = {
    language: string;
    placeholder: string;
    length: number;
}

export interface LocalizationState {
    language: string;
    languages: string[];
}

export enum LocalizationActionType {
    updateLanguage,
    populateLanguages
}

export type LocalizationAction = {
    type: LocalizationActionType;
    payload: {
        language?: string;
        languages?: string[];
    }
}