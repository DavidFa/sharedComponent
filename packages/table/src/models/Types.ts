export type DataGridColumns = {
    field: string;
    headerName: string;
    width: number;
    align?: string;
    sortable?: boolean;
}

export type DataGridRows = {
    [key: string]: string | number
}