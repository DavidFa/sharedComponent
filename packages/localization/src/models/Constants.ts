export const FIELD_NAME: string = "Name";
export const FIELD_DESC: string = "Description";