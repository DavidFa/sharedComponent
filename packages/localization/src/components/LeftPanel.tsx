import React from "react";
import styled from "styled-components";
import InputForm from "./InputForm";
import { FIELD_NAME, FIELD_DESC } from '../models/Constants';

import Message from "./Message";
import Loading from "./Loading";
import useLeftPanel from "./hooks/useLeftPanel";

const Panel = styled.div`
margin: 10px;
padding: 20px;
background: #eee;
box-shadow: 0 6px 20px 10px #ccc;
border-radius: 6px;
`;

const Form = styled.form`
`;

const Button = styled.button`
padding: 5px 10px;
`;

const Wrapper = styled.div`
text-align:right;
`;


const LeftPanel: React.FC = () => {

    const { localization, onShowFlyoutHandler, onChangeFieldHandler, onSubmitHandler } = useLeftPanel();

    return <Panel>
        <Message />
        <Loading />
        <Form data-testid="leftPanelForm" onSubmit={onSubmitHandler}>
            <InputForm label={FIELD_NAME} placeholder={FIELD_NAME} length={20} value={localization.name.tem} showFlyoutHandler={onShowFlyoutHandler} changeField={onChangeFieldHandler} />
            <InputForm label={FIELD_DESC} placeholder={FIELD_DESC} length={50} value={localization.description.tem} showFlyoutHandler={onShowFlyoutHandler} changeField={onChangeFieldHandler} />
            <Wrapper><Button>Save</Button></Wrapper>
        </Form>
    </Panel>
}

export default LeftPanel;