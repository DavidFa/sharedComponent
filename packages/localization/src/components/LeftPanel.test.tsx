import LeftPanel from "./LeftPanel";
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from "react-redux";
import { Status } from '../models/Types';
import { render, screen } from "@testing-library/react";

const mockStore = configureStore([]);

describe("LeftPanel", () => {
    it('should render with message', () => {
        const store = mockStore({
            language: {
                language: "en-AU",
                languages: ["en-AU", "en-US", "en", "en-UK"]
            },
            localization: {
                isShowFlyout: false,
                flyOutField: "Name",
                name: {
                    length: 20,
                    tem: "",
                    localization: {}
                },
                description: {
                    length: 50,
                    tem: "",
                    localization: {}
                }
            },
            message: {
                loading: false,
                status: Status.success,
                message: "success",
            }
        })

        const component = renderer.create(
            <Provider store={store}>
                <LeftPanel />
            </Provider>
        )

        expect(component.toJSON()).toMatchSnapshot();
    })

    it('should render with loading', () => {
        const store = mockStore({
            language: {
                language: "en-AU",
                languages: ["en-AU", "en-US", "en", "en-UK"]
            },
            localization: {
                isShowFlyout: false,
                flyOutField: "Name",
                name: {
                    length: 20,
                    tem: "",
                    localization: {}
                },
                description: {
                    length: 50,
                    tem: "",
                    localization: {}
                }
            },
            message: {
                loading: true,
                status: Status.init,
                message: "",
            }
        })

        const component = renderer.create(
            <Provider store={store}>
                <LeftPanel />
            </Provider>
        )

        expect(component.toJSON()).toMatchSnapshot();
    })

    it('should render without message and loading', () => {
        const store = mockStore({
            language: {
                language: "en-AU",
                languages: ["en-AU", "en-US", "en", "en-UK"]
            },
            localization: {
                isShowFlyout: false,
                flyOutField: "Name",
                name: {
                    length: 20,
                    tem: "",
                    localization: {}
                },
                description: {
                    length: 50,
                    tem: "",
                    localization: {}
                }
            },
            message: {
                loading: false,
                status: Status.init,
                message: "",
            }
        })

        const component = renderer.create(
            <Provider store={store}>
                <LeftPanel />
            </Provider>
        )

        expect(component.toJSON()).toMatchSnapshot();
    })

    describe('LeftPanel without message and loading', () => {
        let store;

        beforeEach(() => {
            store = mockStore({
                language: {
                    language: "en-AU",
                    languages: ["en-AU", "en-US", "en", "en-UK"]
                },
                localization: {
                    isShowFlyout: false,
                    flyOutField: "Name",
                    name: {
                        length: 20,
                        tem: "",
                        localization: {}
                    },
                    description: {
                        length: 50,
                        tem: "",
                        localization: {}
                    }
                },
                message: {
                    loading: false,
                    status: Status.init,
                    message: "",
                }
            })
            store.dispatch = jest.fn();
            render(
                <Provider store={store}>
                    <LeftPanel />
                </Provider>
            )
        })

        it('should contain a form element', () => {
            const form = screen.queryByTestId('leftPanelForm');
            expect(form).toBeInTheDocument();
        })

        it('should contain 2 input elements', () => {
            const inputs = screen.queryAllByRole('textbox');
            expect(inputs).toHaveLength(2);
        })

        it('should contain a button elements', () => {
            const inputs = screen.queryByText('Save')
            expect(inputs).toBeInTheDocument();
        })
    })
    
    describe('LeftPanel with message and loading', () => {
        let store;

        beforeEach(() => {
            store = mockStore({
                language: {
                    language: "en-AU",
                    languages: ["en-AU", "en-US", "en", "en-UK"]
                },
                localization: {
                    isShowFlyout: false,
                    flyOutField: "Name",
                    name: {
                        length: 20,
                        tem: "",
                        localization: {}
                    },
                    description: {
                        length: 50,
                        tem: "",
                        localization: {}
                    }
                },
                message: {
                    loading: true,
                    status: Status.init,
                    message: "success",
                }
            })
            store.dispatch = jest.fn();
            render(
                <Provider store={store}>
                    <LeftPanel />
                </Provider>
            )
        })

        it('should contain a message element', () => {
            const message = screen.queryByText('success');
            expect(message).toBeInTheDocument();
        })

        it('should contain a loading element', () => {
            const inputs = screen.queryByText('Loading', {exact: false});
            expect(inputs).toBeInTheDocument();
        })
    })
})