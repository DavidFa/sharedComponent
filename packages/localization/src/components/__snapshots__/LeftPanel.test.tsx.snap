// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftPanel should render with loading 1`] = `
<div
  className="sc-gKseQn fpshPN"
>
  <div
    className="sc-eCstlR cItJND"
    color=""
  >
    
  </div>
  <div
    className="sc-jSgvzq kQrgyl"
  >
    <div
      className="loading"
    >
      Loading...
    </div>
  </div>
  <form
    className="sc-iBPTik bURNyi"
    data-testid="leftPanelForm"
    onSubmit={[Function]}
  >
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <label
        className="sc-gsTEea fAFGmn"
      >
        Name
        :
      </label>
      <input
        className="sc-hKgJUU kdxHsE"
        maxLength={20}
        onChange={[Function]}
        placeholder="Name"
        value=""
      />
      <button
        className="sc-dlfnuX gmFgaw"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <label
        className="sc-gsTEea fAFGmn"
      >
        Description
        :
      </label>
      <input
        className="sc-hKgJUU kdxHsE"
        maxLength={50}
        onChange={[Function]}
        placeholder="Description"
        value=""
      />
      <button
        className="sc-dlfnuX gmFgaw"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="sc-pGacB eMoSMC"
    >
      <button
        className="sc-fubCzh gKcJWl"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;

exports[`LeftPanel should render with message 1`] = `
<div
  className="sc-gKseQn fpshPN"
>
  <div
    className="sc-eCstlR isDrck"
    color="green"
  >
    success
  </div>
  <div
    className="sc-jSgvzq kQrgyl"
  />
  <form
    className="sc-iBPTik bURNyi"
    data-testid="leftPanelForm"
    onSubmit={[Function]}
  >
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <label
        className="sc-gsTEea fAFGmn"
      >
        Name
        :
      </label>
      <input
        className="sc-hKgJUU kdxHsE"
        maxLength={20}
        onChange={[Function]}
        placeholder="Name"
        value=""
      />
      <button
        className="sc-dlfnuX gmFgaw"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <label
        className="sc-gsTEea fAFGmn"
      >
        Description
        :
      </label>
      <input
        className="sc-hKgJUU kdxHsE"
        maxLength={50}
        onChange={[Function]}
        placeholder="Description"
        value=""
      />
      <button
        className="sc-dlfnuX gmFgaw"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="sc-pGacB eMoSMC"
    >
      <button
        className="sc-fubCzh gKcJWl"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;

exports[`LeftPanel should render without message and loading 1`] = `
<div
  className="sc-gKseQn fpshPN"
>
  <div
    className="sc-eCstlR cItJND"
    color=""
  >
    
  </div>
  <div
    className="sc-jSgvzq kQrgyl"
  />
  <form
    className="sc-iBPTik bURNyi"
    data-testid="leftPanelForm"
    onSubmit={[Function]}
  >
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <label
        className="sc-gsTEea fAFGmn"
      >
        Name
        :
      </label>
      <input
        className="sc-hKgJUU kdxHsE"
        maxLength={20}
        onChange={[Function]}
        placeholder="Name"
        value=""
      />
      <button
        className="sc-dlfnuX gmFgaw"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="sc-bdfBQB cAMYHB"
    >
      <label
        className="sc-gsTEea fAFGmn"
      >
        Description
        :
      </label>
      <input
        className="sc-hKgJUU kdxHsE"
        maxLength={50}
        onChange={[Function]}
        placeholder="Description"
        value=""
      />
      <button
        className="sc-dlfnuX gmFgaw"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="sc-pGacB eMoSMC"
    >
      <button
        className="sc-fubCzh gKcJWl"
      >
        Save
      </button>
    </div>
  </form>
</div>
`;
